aws/resource_aws_cloudhsm2_hsm.go

```diff
@@ -0,0 +1,285 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudhsmv2"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func resourceAwsCloudHsm2Hsm() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudHsm2HsmCreate,
+		Read:   resourceAwsCloudHsm2HsmRead,
+		Update: resourceAwsCloudHsm2HsmUpdate,
+		Delete: resourceAwsCloudHsm2HsmDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsCloudHsm2HsmImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(120 * time.Minute),
+			Update: schema.DefaultTimeout(120 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"hsm_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hsm_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hsm_eni_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudHsm2HsmImport(
```

Comment:
Since this resource also requires `cluster_id` to properly perform its `Read` function, we need to accommodate for this here as well. We have a choice, we can either setup the import to use a different ID than the resource (e.g. `CLUSTERID/HSMID`) and add that parsing here to properly `d.Set("cluster_id", ...)` or we should add the cluster ID to the resource's ID (e.g. `d.SetId(fmt.Sprintf("%s/%s", clusterID, hsmID)`), parse `d.Id()` for wherever necessary in the `Read`, `Delete`, and testing functions, and use `schema.ImportStatePassThrough` instead.